examples/v1alpha2/basic-tcp.yaml

```diff
@@ -0,0 +1,45 @@
+kind: Gateway
+apiVersion: gateway.networking.k8s.io/v1alpha2
```

Comment:
Recommend swapping order of these fields to align with https://github.com/kubernetes-sigs/gateway-api/issues/810

Reply:
3420bb08ada03b6e7cba3787286c9fd0c8b18cd5 changes the provided example YAML to conform with #810 :+1: